### PR TITLE
Mule 9680: Review tests for new TCP/UDP/SSL connectors

### PR DESCRIPTION
**Pull request by pablomusumeci**

### CODE REVIEW

File: extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@SmallTest
+@RunWith(MockitoJUnitRunner.class)
+public class TcpReconnectionTestCase extends AbstractMuleTestCase
+{
+
+    @Inject
+    MuleContext muleContext;
```
Comment (marianogonzalez) [inline extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java]: private

File: extensions/sockets/pom.xml [hunk 2]
```diff
@@ -33,6 +33,11 @@
 
     <dependencies>
         <dependency>
+            <groupId>commons-io</groupId>
```
Comment (marianogonzalez) [inline extensions/sockets/pom.xml]: if you do this then you have to exclude it on the all-extensions pom

File: extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Rule
+    public DynamicPort dynamicPort = new DynamicPort("port");
```
Comment (marianogonzalez) [inline extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java]: per conventions, declare the public rules first

File: extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @Test
+    public void closeListener() throws Exception
+    {
+        expectedException.expect(SocketException.class);
+        expectedException.expectMessage("Socket is closed");
+        listenerConnection.connect();
+
+        new Runnable()
```
Comment (marianogonzalez) [inline extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java]: what's this runnable for? this is the same as calling listenerConnection.disconnect(); directly

Comment (pablomusumeci) [inline extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java]: listenerConnection.listen is blocking, so we need to disconnect it from a different thread

Comment (marianogonzalez) [inline extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java]: but you're not doing it in a different thread. You're running it in the current thread

Comment (marianogonzalez) [inline extensions/sockets/src/test/java/org/mule/extension/socket/TcpReconnectionTestCase.java]: besides if you were doing it in a different thread, you need a latch to control race conditions. This test is not doing what you say it does, and if it did, it'd be flaky

**Pull request merged**